hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/compact/CompactionPlanOperator.java

```diff
@@ -81,6 +83,11 @@ public void open() throws Exception {
     // these instants are in priority for scheduling task because the compaction instants are
     // scheduled from earliest(FIFO sequence).
     CompactionUtil.rollbackCompaction(table);
+
+    // try doing compcation once for batch mode
+    if (OptionsResolver.isMorTable(conf) && !conf.getBoolean(FlinkOptions.COMPACTION_ASYNC_ENABLED)) {
+      this.notifyCheckpointComplete(-1);
```

Comment:
If compaction is disabled, there might not be compaction sub-pipeline in the workload now.